clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -5903,7 +5904,7 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
       auto Call = RValue::get(
           EmitRuntimeCall(CGM.CreateRuntimeFunction(FTy, Name), Args));
       if (TmpSize)
-        EmitLifetimeEnd(TmpSize, TmpPtr);
+        EmitLifetimeEnd(TmpSize, TmpPtr->stripPointerCasts());
```

Comment:
I'm surprised CreateMemTemp isn't returning the raw alloca. I would expect it to return the direct reference, and not the addrspacecasted one. 

Assuming fixing that is hard, should just do the strip once and use it consistently

Reply:
I would extremely strongly prefer not to mess with this ball of yarn or other pieces of functionality (specifically, `CreateMemTemp`), as part of what is essentially a point fix of a long standing bug. I'll leave a TODO in place for someone that is braver and more skilled.

Reply:
Can still just do the stripPointerCasts once above